runtime/jvmti/jvmtiHook.c

```diff
@@ -421,6 +423,70 @@ jvmtiHookVirtualThreadDestroy(J9HookInterface **hook, UDATA eventNum, void *even
 
 	TRACE_JVMTI_EVENT_RETURN(jvmtiHookVirtualThreadDestroy);
 }
+
+static void
+jvmtiHookVirtualThreadMount(J9HookInterface **hook, UDATA eventNum, void *eventData, void *userData)
+{
+	J9VMVirtualThreadMountEvent *data = eventData;
+	J9JVMTIEnv *j9env = userData;
+	J9VMThread *currentThread = data->currentThread;
+	//jvmtiEventThreadStart callback = j9env->callbacks.VirtualThreadStart;
+	jvmtiExtensionEvent callback = *J9JVMTI_EXTENSION_CALLBACK(j9env, J9JVMTI_EVENT_COM_IBM_VIRTUAL_THREAD_MOUNT);
+
+	Trc_JVMTI_jvmtiHookVirtualThreadMount_Entry();
+
+	ENSURE_EVENT_PHASE_LIVE(jvmtiHookVirtualThreadMount, j9env);
+
+	/* Call the event callback */
+
+	if (NULL != callback) {
+		jthread threadRef = NULL;
+		UDATA hadVMAccess = 0;
+		UDATA javaOffloadOldState = 0;
+		if (prepareForEvent(
+			j9env, currentThread, currentThread, J9JVMTI_EVENT_COM_IBM_VIRTUAL_THREAD_MOUNT,
+			&threadRef, &hadVMAccess, FALSE, 0, &javaOffloadOldState)
+		) {
+			callback((jvmtiEnv *)j9env, (JNIEnv *)currentThread, threadRef);
+			finishedEvent(currentThread, J9JVMTI_EVENT_COM_IBM_VIRTUAL_THREAD_MOUNT, hadVMAccess, javaOffloadOldState);
+		}
+	}
+
+	TRACE_JVMTI_EVENT_RETURN(jvmtiHookVirtualThreadMount);
+}
+
+static void
+jvmtiHookVirtualThreadUnmount(J9HookInterface **hook, UDATA eventNum, void *eventData, void *userData)
+{
+	J9VMVirtualThreadUnmountEvent *data = eventData;
+	J9JVMTIEnv *j9env = userData;
+	//jvmtiEventThreadEnd callback = j9env->callbacks.VirtualThreadEnd;
+	jvmtiExtensionEvent callback = *J9JVMTI_EXTENSION_CALLBACK(j9env, J9JVMTI_EVENT_COM_IBM_VIRTUAL_THREAD_UNMOUNT);
+	J9VMThread *currentThread = data->currentThread;
+
+	Trc_JVMTI_jvmtiHookVirtualThreadUnmount_Entry();
+
+	ENSURE_EVENT_PHASE_LIVE(jvmtiHookVirtualThreadUnmount, j9env);
+
+	/* Call the event callback */
+
+	if (NULL != callback) {
+		jthread threadRef = NULL;
+		UDATA hadVMAccess = 0;
+		UDATA javaOffloadOldState = 0;
+
+		if (prepareForEvent(
+			j9env, currentThread, currentThread, J9JVMTI_EVENT_COM_IBM_VIRTUAL_THREAD_UNMOUNT,
+			&threadRef, &hadVMAccess, FALSE, 0, &javaOffloadOldState)
```

Comment:
The last third parameter should be `TRUE` (= `wantVMAccess`). Setting it to FALSE will relinquish VM access and cause hangs/deadlocks which are currently seen in the local runs.

Reply:
changed the last third parameter from FALSE to TRUE as suggested.